hydromt/components/dataset.py

```diff
@@ -0,0 +1,405 @@
+"""Xarrays component."""
+
+from pathlib import Path
+from shutil import move
+from typing import (
+    TYPE_CHECKING,
+    Dict,
+    List,
+    Optional,
+    Union,
+    cast,
+)
+
+from pandas import DataFrame
+from xarray import DataArray, Dataset
+
+from hydromt._typing.type_def import XArrayDict
+from hydromt.components.base import ModelComponent
+from hydromt.hydromt_step import hydromt_step
+from hydromt.io.readers import read_nc
+from hydromt.io.writers import write_nc
+
+if TYPE_CHECKING:
+    from hydromt.models.model import Model
+
+_DEFAULT_DATASET_FILENAME = "{name}.nc"
```

Comment:
I did not do it because we can't give multiple folders here such as `forcings` as it's a generic component but if you think it's better I'm happy to add one.